runner/database.go

```diff
@@ -508,6 +508,26 @@ func (ec EvalContext) EvalDatabasePanel(
 
 		wroteFirstRow := false
 		return withJSONArrayOutWriterFile(w, func(w *jsonutil.StreamEncoder) error {
+			if ec.settings.CacheMode {
+				rows, err := db.Queryx(query)
+				if err != nil {
+					return err
+				}
+
+				defer rows.Close()
+
+				for rows.Next() {
+					err := writeRowFromDatabase(dbInfo, w, rows, wroteFirstRow)
+					if err != nil {
+						return err
+					}
+
+					wroteFirstRow = true
+				}
+
+				return rows.Err()
+			}
```

Comment:
Ok no I wouldn't do it like this. But good idea!

[Here](https://github.com/multiprocessio/dsq/blob/main/main.go#L354) is where dsq creates the in memory file. Instead of using that in-memory tmp file you should create a real file for it there.

I wonder what the perf difference is.

I don't think we should modify datastation/runner code almost at all for this. Maybe we can modify it to return a non-tmp sqliteconnector. So the line in dsq could be

```
connector, err := runner.MakeTmpSQLiteConnector()
if cache {
  connector, err = runner.MakePermanentSQLiteConnector(inputFileNamesHash)
}
```

Does that make sense?

Reply:
That's exactly what I did here https://github.com/multiprocessio/dsq/pull/46.
If we don't modify the runner code, then wouldn't the previous data be overwritten by the latest run?

Reply:
Ah it's that when cached mode is on we should completely skip the import step: https://github.com/multiprocessio/dsq/blob/a79464c5593e47ca69c64745a4cbb8b90767cd78/main.go#L335. This whole loop should be skipped when cached is set.

Because we can just load directly from the cached database that already has all the rows in it. Right?

Reply:
Oh yeah, why didn't I think of that :)